src/node_api.cc

```diff
@@ -328,12 +329,18 @@ class CallbackWrapperBase : public CallbackWrapper {
         v8::Local<v8::External>::Cast(
             _cbdata->GetInternalField(kInternalFieldIndex))->Value());
     v8::Isolate* isolate = _cbinfo.GetIsolate();
-    cb(v8impl::JsEnvFromV8Isolate(isolate), cbinfo_wrapper);
+    napi_env env = static_cast<napi_env>(
+        v8::Local<v8::External>::Cast(
+            _cbdata->GetInternalField(kEnvIndex))->Value());
+
+    // Make sure any errors encountered last time we were in N-API are gone.
+    napi_clear_last_error(env);
```

Comment:
I'll add a test for https://gist.github.com/gabrielschulhof/a01fa89e98ce1042dd6ec6776eada3b4